src/plugins/plugin.tooltip.js

```diff
@@ -350,6 +350,102 @@ function overrideCallbacks(callbacks, context) {
   return override ? callbacks.override(override) : callbacks;
 }
 
+const defaultCallbacks = {
+  // Args are: (tooltipItems, data)
+  beforeTitle: noop,
+  title(tooltipItems) {
+    if (tooltipItems.length > 0) {
+      const item = tooltipItems[0];
+      const labels = item.chart.data.labels;
+      const labelCount = labels ? labels.length : 0;
+
+      if (this && this.options && this.options.mode === 'dataset') {
+        return item.dataset.label || '';
+      } else if (item.label) {
+        return item.label;
+      } else if (labelCount > 0 && item.dataIndex < labelCount) {
+        return labels[item.dataIndex];
+      }
+    }
+
+    return '';
+  },
+  afterTitle: noop,
+
+  // Args are: (tooltipItems, data)
+  beforeBody: noop,
+
+  // Args are: (tooltipItem, data)
+  beforeLabel: noop,
+  label(tooltipItem) {
+    if (this && this.options && this.options.mode === 'dataset') {
+      return tooltipItem.label + ': ' + tooltipItem.formattedValue || tooltipItem.formattedValue;
+    }
+
+    let label = tooltipItem.dataset.label || '';
+
+    if (label) {
+      label += ': ';
+    }
+    const value = tooltipItem.formattedValue;
+    if (!isNullOrUndef(value)) {
+      label += value;
+    }
+    return label;
+  },
+  labelColor(tooltipItem) {
+    const meta = tooltipItem.chart.getDatasetMeta(tooltipItem.datasetIndex);
+    const options = meta.controller.getStyle(tooltipItem.dataIndex);
+    return {
+      borderColor: options.borderColor,
+      backgroundColor: options.backgroundColor,
+      borderWidth: options.borderWidth,
+      borderDash: options.borderDash,
+      borderDashOffset: options.borderDashOffset,
+      borderRadius: 0,
+    };
+  },
+  labelTextColor() {
+    return this.options.bodyColor;
+  },
+  labelPointStyle(tooltipItem) {
+    const meta = tooltipItem.chart.getDatasetMeta(tooltipItem.datasetIndex);
+    const options = meta.controller.getStyle(tooltipItem.dataIndex);
+    return {
+      pointStyle: options.pointStyle,
+      rotation: options.rotation,
+    };
+  },
+  afterLabel: noop,
+
+  // Args are: (tooltipItems, data)
+  afterBody: noop,
+
+  // Args are: (tooltipItems, data)
+  beforeFooter: noop,
+  footer: noop,
+  afterFooter: noop
+};
+
+/**
+ * Invoke callback from object with contextn and arguments.
```

Comment:
typo in context

Reply:
@kurkle Now this fallback is not used, should I save it?